ocaml/src/asd_server.ml

```diff
@@ -217,24 +218,39 @@ module DirectoryInfo = struct
             (* TODO push to blob module? *)
             (match blob with
              | Lwt_bytes s ->
+                let len = Lwt_bytes.length s in
                 Lwt_extra2.write_all_lwt_bytes
                   fd
-                  s 0 (Lwt_bytes.length s)
+                  s 0 len
+                >>= fun () ->
+                Lwt.return len
+                  
              | Bigslice s ->
                 let open Bigstring_slice in
+                let len = s.length in
                 Lwt_extra2.write_all_lwt_bytes
                   fd
-                  s.bs s.offset s.length
+                  s.bs s.offset s.length >>= fun () ->
+                Lwt.return len
              | Bytes s ->
+                let len = Bytes.length s in
                 Lwt_extra2.write_all
                   fd
-                  s 0 (Bytes.length s)
+                  s 0 len
+                >>= fun () ->
+                Lwt.return len
              | Slice s ->
                 let open Slice in
+                let len = s.length in
                 Lwt_extra2.write_all
                   fd
-                  s.buf s.offset s.length
-            ) >>= fun () ->
+                  s.buf s.offset len
+                >>= fun () ->
+                Lwt.return len
+            )
+            >>= fun length ->
+            Posix.lwt_posix_fadvise fd 0 length Posix.POSIX_FADV_DONTNEED
```

Comment:
This fadvise should be done after syncing the blobs to disk. The sync is done in `post_write` 2 lines below.